src/Webhooks.API/Program.cs

```diff
@@ -1,18 +1,18 @@
 ﻿var builder = WebApplication.CreateBuilder(args);
 
 builder.AddServiceDefaults();
-builder.AddDefaultOpenApi();
 builder.AddApplicationServices();
 
-var app = builder.Build();
+var apiVersioning = builder.Services.AddApiVersioning();
 
-app.UseDefaultOpenApi();
+builder.AddDefaultOpenApi(apiVersioning);
```

Comment:
Is there a reason we have to pass the `IApiVersioningBuilder` via the arguments in this way? It seems to me that we could possible call `AddApiVersioning` inside the `AddDefaultOpenApi` call.

Reply:
The reason that `IApiVersioningBuilder` is passed here has to do with how API Versioning is configured. As an example, `IMvcCoreBuilder.AddApiExplorer` is different than `IApiVersioningBuilder.AddApiExplorer`, but they intentionally have the same name as they express the same intention. Under the hood, the later calls the former.

It would be possible to **not** pass the `IApiVersioningBuilder` and have a setup inside `AddDefaultOpenApi` that looks like:

```c#
builder.Services.AddApiVersioning().AddApiExplorer();
```

There's a few reasons why I chose not to do that:

1. Ordering can matter. If `AddApiVersioning()` hasn't already been called and configured, then services _might_ not be configured as expected. It's an edge case not likely to happen in a reference application, but someone copying and modifying it _could_ run into that problem in a very non-obvious way.
2. The name `AddDefaultOpenApi` implies that is configuring OpenAPI. If it also adds API Versioning, that is neither self-describing nor intrinsically obvious.
3. `AddDefaultOpenApi` is a shared extension method. It's possible that different APIs might configure API Versioning different ways; for example, different styles, different defaults, etc. It's better to the keep the API Versioning configuration separate so that it is obvious what and when it is configured. The API Versioning extensions for OpenAPI tend to be the same or overlap for all APIs so keeping it inside `AddDefaultOpenApi` makes sense. It may not even be invoked. By passing `IApiVersioningBuilder`, the ordering problem is solved because you had to have called `AddApiVersioning()` at least once.

I'll let you cogitate on that and decide if you'd like to change it.

Reply:
I've restructured this just a bit to make it read more clearly and retain the original functionality.

### Adding OpenAPI with API Versioning

```c#
var withApiVersioning = builder.Services.AddApiVersioning();
builder.AddDefaultOpenApi(withApiVersioning);
```

### Adding OpenAPI without API Versioning (Original)

```c#
builder.AddDefaultOpenApi();
```

Reply:
@commonsensesoftware Thanks for sharing this! I have a better understanding of the problem here after taking a look at what it would take to integrate Asp.Versioning with the new built-in support in Microsoft.AspNetCore.OpenApi. We can discuss this further next week but for now I appreciate why this is setup this way.